astroid/inference.py

```diff
@@ -368,6 +368,10 @@ def infer_subscript(self, context=None):
         if value is util.Uninferable:
             yield util.Uninferable
             return None
+        if isinstance(value, nodes.ClassDef):
+            if value.is_subtype_of("typing.Generic"):
```

Comment:
Can you also pass the `context` to `is_subtype_of`? This should prevent potential circular inference that could cause `RecursionErrors` to pop up.